### DISCUSSION TITLE
tar_make not building all targets and failing

### DISCUSSION THREAD

**Discussion by bastian-wur** [ropensci/targets#933] [Help] [answered]
Hi everyone,

I hope someone here will be able to help me, because I'm not getting to the source of my problem. It needs to be said that I'm really not an R pro, which might have something to do with this.

The issue:
I am right now modifying a pipeline which a colleague built. Then I'm executing tar_make(), but the pipeline fails. One of the targets should have written a file to drive, the next one then should read it, and it fails because the file is not written.
If I use tar_read on the target, then it comes back empty.
I can call tar_make on only the target, and it will be built.
So now I'm confused. Why are these targets not executed, if they can be executed properly. I'm a bit lost, I'm not getting any further.
Does anyone have an idea?

Thanks, and best regards,
Bastian

**Comment by gorkang** [comment]
Could you post a reproducible example, or at the very least, a copy of the messages you see in the Console?

**Comment by bastian-wur** [comment]
I'll try a reproducible example, but the first thing not working is around line 200, and the final error is thrown around line 500, so... er... I'm struggling.

The error message is:
✖ error target significants_3D2D
• end pipeline: 6.34 seconds
Error : 'results_XXX_3D_vs_2D.tsv' does not exist in current working directory ('D:/MyName/rnasqtest/XXX').
✖ Problem with the pipeline.
Error:
! problem with the pipeline

If I do tar_make(path_results_XXX_3D_vs_2D), which produces the tsv file (from multiple targets before), then this step will work fine, and the file will written do disk, but will then break down at a step later where basically the same things are executed with different inputs.

Deleting the .targets subfolder doesn't change a thing either :/.

**Comment by bastian-wur** [comment, marked as answer]
Oh my, now I feel stupid.

The latter function was directly referencing the file name "results_XXX_3D_vs_2D.tsv" instead of the target "path_results_XXX_3D_vs_2D". Since the target was actually not used anywhere else in the code, the target wasn't built, and therefore the latter function crashed.
I mean... obvious in hindsight.

So, sorry, and disregard, this was me being unfamiliar with the whole environment.
